Makefile

```diff
@@ -27,117 +88,52 @@ httpstan/%_pb2.py: protos/%.proto httpstan/bin/protoc
-
-################################################################################
-# SUNDIALS build rules
-# Note: Files starting with f* are by SUNDIALS convention files needed for
-#       Fortran bindings which we do not need for stan-math. Thus these targets
-#       are ignored here. This convention was introduced with 4.0.
-##
-
-SUNDIALS_CVODES := $(patsubst %.c,%.o,\
-  $(wildcard $(SUNDIALS)/src/cvodes/*.c) \
-  $(wildcard $(SUNDIALS)/src/sundials/*.c) \
-  $(wildcard $(SUNDIALS)/src/sunmatrix/band/[^f]*.c) \
-  $(wildcard $(SUNDIALS)/src/sunmatrix/dense/[^f]*.c) \
-  $(wildcard $(SUNDIALS)/src/sunlinsol/band/[^f]*.c) \
-  $(wildcard $(SUNDIALS)/src/sunlinsol/dense/[^f]*.c) \
-  $(wildcard $(SUNDIALS)/src/sunnonlinsol/newton/[^f]*.c) \
-  $(wildcard $(SUNDIALS)/src/sunnonlinsol/fixedpoint/[^f]*.c))
-
-SUNDIALS_IDAS := $(patsubst %.c,%.o,\
-  $(wildcard $(SUNDIALS)/src/idas/*.c) \
-  $(wildcard $(SUNDIALS)/src/sundials/*.c) \
-  $(wildcard $(SUNDIALS)/src/sunmatrix/band/[^f]*.c) \
-  $(wildcard $(SUNDIALS)/src/sunmatrix/dense/[^f]*.c) \
-  $(wildcard $(SUNDIALS)/src/sunlinsol/band/[^f]*.c) \
-  $(wildcard $(SUNDIALS)/src/sunlinsol/dense/[^f]*.c) \
-  $(wildcard $(SUNDIALS)/src/sunnonlinsol/newton/[^f]*.c) \
-  $(wildcard $(SUNDIALS)/src/sunnonlinsol/fixedpoint/[^f]*.c))
-
-SUNDIALS_NVECSERIAL := $(patsubst %.c,%.o,\
-  $(addprefix $(SUNDIALS)/src/, nvector/serial/nvector_serial.c sundials/sundials_math.c))
-
-$(sort $(SUNDIALS_CVODES) $(SUNDIALS_IDAS) $(SUNDIALS_NVECSERIAL)) : CXXFLAGS = $(CXXFLAGS_SUNDIALS) $(CXXFLAGS_OS) -O$(O) $(INC_SUNDIALS)
-$(sort $(SUNDIALS_CVODES) $(SUNDIALS_IDAS) $(SUNDIALS_NVECSERIAL)) : CPPFLAGS = $(CPPFLAGS_SUNDIALS) $(CPPFLAGS_OS)
-$(sort $(SUNDIALS_CVODES) $(SUNDIALS_IDAS) $(SUNDIALS_NVECSERIAL)) : %.o : %.c
-	@mkdir -p $(dir $@)
-	$(COMPILE.cpp) -x c -include $(SUNDIALS)/include/stan_sundials_printf_override.hpp $< $(OUTPUT_OPTION)
-
-$(SUNDIALS)/lib/libsundials_cvodes.a: $(SUNDIALS_CVODES)
-	@mkdir -p $(dir $@)
-	$(AR) -rs $@ $^
-
-$(SUNDIALS)/lib/libsundials_idas.a: $(SUNDIALS_IDAS)
-	@mkdir -p $(dir $@)
-	$(AR) -rs $@ $^
-
-$(SUNDIALS)/lib/libsundials_nvecserial.a: $(SUNDIALS_NVECSERIAL)
-	@mkdir -p $(dir $@)
-	$(AR) -rs $@ $^
-
-LIBSUNDIALS := $(SUNDIALS)/lib/libsundials_nvecserial.a $(SUNDIALS)/lib/libsundials_cvodes.a $(SUNDIALS)/lib/libsundials_idas.a
-
-STAN_SUNDIALS_HEADERS := $(shell find $(MATH)stan -name *cvodes*.hpp) $(shell find $(MATH)stan -name *idas*.hpp)
-$(STAN_SUNDIALS_HEADERS) : $(LIBSUNDIALS)
-
-clean-sundials:
-	@echo '  cleaning sundials targets'
-	$(RM) $(wildcard $(sort $(SUNDIALS_CVODES) $(SUNDIALS_IDAS) $(SUNDIALS_NVECSERIAL) $(LIBSUNDIALS)))
-
+httpstan/lib/%: build/math-$(MATH_VERSION)/lib/sundials_$(SUNDIALS_VERSION)/lib/%
+	cp $< $@
 
 ###############################################################################
-# httpstan-specific libsundials
+# Build Stan-related shared libraries using Makefile.libraries
```

Comment:
Looks good. I don't fully grasp what is going on here, but if it works, then it works.

Reply:
I can improve the comment. `Makefile.libraries` is, essentially, a simplified version of stan math's makefile.

Reply:
Added more comments to explain that the new `Makefile.libraries` is a trimmed version of Stan Math's `makefile`.